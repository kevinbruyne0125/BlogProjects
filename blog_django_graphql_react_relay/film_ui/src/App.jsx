import React, { Component } from "react";
import { PageHeader } from "react-bootstrap";

import FilmList from "./FilmList";

class App extends Component {
  render() {
    return (
      <div>
        <PageHeader style={{ "text-align": "center" }}>
          Films database
        </PageHeader>
        <FilmList />
      </div>
    );
  }
}

export default App;
