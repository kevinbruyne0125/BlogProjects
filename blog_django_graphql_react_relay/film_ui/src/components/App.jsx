import React, { Component } from "react";
import { QueryRenderer, graphql } from "react-relay";
import environment from "../Environment";

import FilmList from "./FilmList";

const FilmListQuery = graphql`
  query AppQuery {
    films {
      id
      ...FilmList_films
    }
  }
`;

class App extends Component {
  render() {
    return (
      <QueryRenderer
        environment={environment}
        query={FilmListQuery}
        render={({ error, props }) => {
          if (error) {
            return <div>{error.message}</div>;
          } else if (props) {
            return <FilmList films={props.films} />;
          }
          return <div>Loading</div>;
        }}
      />
    );
  }
}

export default App;
