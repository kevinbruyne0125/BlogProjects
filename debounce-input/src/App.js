import React, { Component } from "react";

class App extends Component {
  render() {
    return (
      <div className="flex flex-col items-center h-screen w-full bg-teal-lighter bg-repeat">
        <div className="container md:max-w-sm md:mx-auto">
          <h1 className="block w-full text-center text-grey-darkest mb-6">
            Debounce in React
          </h1>
          <form className="mb-4">
            <div className="flex flex-col mb-4 md:w-full">
              <label
                className="mb-2 uppercase font-bold text-lg text-grey-darkest"
                htmlFor="search-input"
              >
                Search input:
              </label>
              <input className="field" name="search" type="text" id="search" />
            </div>
          </form>
        </div>
        <div className="container md:max-w-sm md:mx-auto">
          <span>Your typed characters</span>
        </div>
      </div>
    );
  }
}

export default App;
