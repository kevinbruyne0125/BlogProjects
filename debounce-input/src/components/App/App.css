@tailwind preflight;

/* Here we can add any custom overrides */
.field {
  @apply .border .py-2 .px-3 .text-grey-darkest;
}

.container {
  @apply .w-full .bg-white .rounded .shadow-lg .p-8 .m-4;
}

@tailwind utilities;

/* Here we can add our own utilities or custom CSS */
