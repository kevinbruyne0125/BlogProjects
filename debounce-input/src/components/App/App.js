import React, { Component } from "react";
import debounce from "lodash.debounce";

import SearchInput from "../SearchInput/SearchInput";
import SearchResult from "../SearchResult/SearchResult";

class App extends Component {
  constructor(props) {
    super(props);

    this.emitChangeDebounced = debounce(this.emitChange, 250);
  }

  componentWillUnmount() {
    this.emitChangeDebounced.cancel();
  }

  handleChange = event => {
    this.emitChangeDebounced(event.target.value);
  };

  emitChange = value => {
    if (value !== "") {
      this.props.addWord(value);
    }
  };

  render() {
    return (
      <div className="flex flex-col items-center min-h-screen w-full bg-teal-lighter bg-repeat">
        <div className="container md:max-w-sm md:mx-auto">
          <h1 className="block w-full text-center text-grey-darkest mb-6">
            Debounce in React
          </h1>
          <SearchInput handleChange={this.handleChange} />
        </div>
        {this.props.words.map((word, key) => (
          <SearchResult text={word} key={key} />
        ))}
      </div>
    );
  }
}

export default App;
