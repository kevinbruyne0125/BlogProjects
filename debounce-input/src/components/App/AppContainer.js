import React from "react";
import { connect } from "react-redux";

import { addWord } from "../../actions/index";

import App from "./App";

export const AppContainer = props => (
  <App addWord={props.addWord} words={props.words} />
);

const mapDispatchToProps = dispatch => ({
  addWord: word => dispatch(addWord(word))
});

const mapStateToProps = state => ({
  words: state.words
});

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
