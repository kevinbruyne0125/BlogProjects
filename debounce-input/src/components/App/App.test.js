import React from "react";

import { shallow } from "enzyme";

import App from "./App";

describe("App Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<App addWord={jest.fn()} words={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
