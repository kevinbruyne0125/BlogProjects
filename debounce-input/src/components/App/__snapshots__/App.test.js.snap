// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addWord={[Function]}
    words={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container md:max-w-sm md:mx-auto"
>
          <h1
                    className="block w-full text-center text-grey-darkest mb-6"
          >
                    Debounce in React
          </h1>
          <SearchInput
                    handleChange={[Function]}
          />
</div>,
        Array [],
      ],
      "className": "flex flex-col items-center min-h-screen w-full bg-teal-lighter bg-repeat",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="block w-full text-center text-grey-darkest mb-6"
>
              Debounce in React
</h1>,
            <SearchInput
              handleChange={[Function]}
/>,
          ],
          "className": "container md:max-w-sm md:mx-auto",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Debounce in React",
              "className": "block w-full text-center text-grey-darkest mb-6",
            },
            "ref": null,
            "rendered": "Debounce in React",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container md:max-w-sm md:mx-auto"
>
            <h1
                        className="block w-full text-center text-grey-darkest mb-6"
            >
                        Debounce in React
            </h1>
            <SearchInput
                        handleChange={[Function]}
            />
</div>,
          Array [],
        ],
        "className": "flex flex-col items-center min-h-screen w-full bg-teal-lighter bg-repeat",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="block w-full text-center text-grey-darkest mb-6"
>
                Debounce in React
</h1>,
              <SearchInput
                handleChange={[Function]}
/>,
            ],
            "className": "container md:max-w-sm md:mx-auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Debounce in React",
                "className": "block w-full text-center text-grey-darkest mb-6",
              },
              "ref": null,
              "rendered": "Debounce in React",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
